Python/tests/test_rerfClassifier.py

```diff
@@ -77,6 +78,71 @@ def test_attributes():
     assert np.allclose(pred_proba, np.exp(pred_log_proba))
 
 
+def test_alt_attributes():
+    clf = rerfClassifier(n_jobs=3, max_features=4)
+
+    clf.fit(iris.data, iris.target)
+
+    assert hasattr(clf, "n_jobs_")
+    assert hasattr(clf, "mtry_")
+
+    assert clf.n_jobs_ == 3
+    assert clf.mtry_ == 4
+
+
+def prep_digit_data():
+    from sklearn import datasets, metrics
+
+    digits = datasets.load_digits()
+    # The data that we are interested in is made of 8x8 images of digits, let's
+    # have a look at the first 4 images, stored in the `images` attribute of the
+    # dataset.  If we were working from image files, we could load them using
+    # matplotlib.pyplot.imread.  Note that each image must have the same size. For these
+    # images, we know which digit they represent: it is given in the 'target' of
+    # the dataset.
+
+    # To apply a classifier on this data, we need to flatten the image, to
+    # turn the data in a (samples, feature) matrix:
+    n_samples = len(digits.images)
+    data = digits.images.reshape((n_samples, -1))
+
+    return data, digits.target, n_samples
+
+
+def test_s_rerf():
+    # asserts that we can at least run s-rerf on some image data
+    X, y, n = prep_digit_data()
+    X_train = X[: n // 2]
+    y_train = y[: n // 2]
+
+    clf = rerfClassifier(
+        projection_matrix="S-RerF", image_height=8, image_width=8, n_estimators=10
+    )
+
+    clf.fit(X_train, y_train)
+
+    x_test = X[n // 2 :]
+    y_test = y[n // 2 :]
+
+    score = clf.score(x_test, y_test)
+
+    assert score > 0.75
+
+    assert hasattr(clf, "image_height")
+    assert hasattr(clf, "image_width")
+    assert hasattr(clf, "patch_width_max_")
+    assert hasattr(clf, "patch_width_min")
+    assert hasattr(clf, "patch_height_max_")
+    assert hasattr(clf, "patch_height_min")
+
+    assert clf.image_height == 8
+    assert clf.image_width == 8
+    assert clf.patch_width_max_ == 8
+    assert clf.patch_width_min == 1
+    assert clf.patch_height_max_ == 8
```

Comment:
some of these have underscores and some do not...  is there a reason for that?

Reply:
Fixed.